FEM/burgers.cpp

```diff
@@ -0,0 +1,152 @@
+    B = (*data)(11); // constant factor for Arrhenius term
+    Q = (*data)(12); // activation energy in Arrhenius term
+
+    GM = GM0;
+    KM = KM0;
+    GK = GK0;
+    etaK = etaK0;
+    etaM = etaM0;
+
+    smath = new Invariants();
+}
+
+SolidBurgers::~SolidBurgers()
+{
+    smath = NULL;
+}
+
+/**************************************************************************
+   FEMLib-Method: Burgers::UpdateBurgersProperties()
+   Task: Updates BURGERS material parameters in LUBBY2 fashion
+   Programing:
+   07/2014 TN Implementation
+**************************************************************************/
+void SolidBurgers::UpdateBurgersProperties(const double s_eff, const double Temperature)
+{
+	double dT = Temperature - T_ref;
+	GM = GM0 + m_GM*dT;
+	KM = KM0 + m_KM*dT;
+	GK = GK0 * std::exp(mK * s_eff);
+	etaK = etaK0 * std::exp(mvK * s_eff);
+    etaM = etaM0 * std::exp(mvM * s_eff) * B * std::exp(Q/(PhysicalConstant::IdealGasConstant*Temperature));
+	if (etaM/etaM0 < 1.e-2)
+		std::cout << "WARNING: Maxwell viscosity sank to 100th of original value." << std::endl;
+}
+
+/**************************************************************************
+   FEMLib-Method: Burgers::CalResidualBurgers()
+   Task: Calculates the 12x1 residual vector. Implementation fully implicit only.
+   Programing:
+   06/2014 TN Implementation
+**************************************************************************/
+void SolidBurgers::CalResidualBurgers(const double dt, const Eigen::Matrix<double,6,1> &strain_curr, const Eigen::Matrix<double,6,1> &stress_curr,
+                                      Eigen::Matrix<double,6,1> &strain_Kel_curr, const Eigen::Matrix<double,6,1> &strain_Kel_t,
+                                      Eigen::Matrix<double,6,1> &strain_Max_curr, const Eigen::Matrix<double,6,1> &strain_Max_t, Eigen::Matrix<double,18,1> &res)
+{
+    //calculate stress residual
+    res.block<6,1>(0,0) = stress_curr - 2. * (strain_curr - strain_Kel_curr - strain_Max_curr);
+
+    //calculate Kelvin strain residual
+    res.block<6,1>(6,0) = 1./dt * (strain_Kel_curr - strain_Kel_t) - 1./(2.*etaK) * (GM*stress_curr - 2.*GK*strain_Kel_curr);
+
+    //calculate Maxwell strain residual
+    res.block<6,1>(12,0) = 1./dt * (strain_Max_curr - strain_Max_t) - GM/(2.*etaM)*stress_curr;
+
+
+}
+
+/**************************************************************************
+   FEMLib-Method: Burgers::CalJacobianBurgers()
+   Task: Calculates the 12x12 Jacobian. Implementation fully implicit only.
+   Programing:
+   06/2014 TN Implementation
+**************************************************************************/
+void SolidBurgers::CalJacobianBurgers(const double dt, Eigen::Matrix<double,18,18> &Jac, const double s_eff, const Eigen::Matrix<double,6,1> &sig_i, const Eigen::Matrix<double,6,1> &eps_K_i)
+{
+    //Check Dimension of Jacobian
+    if (Jac.cols() != 18 || Jac.rows() != 18)
```

Comment:
do you really need this check? You fix the size of Jac in `Eigen::Matrix<double,18,18> &Jac`...